### PR TITLE
Link all platforms

### PR DESCRIPTION
**Pull request by rozele**
## Motivation

@grabbou requested that we eliminate the special cases for iOS and Android in `link` and `unlink` commands.

## Test Plan

- Ensure `jest` tests continue to work.
- Link and unlink a module for an iOS project without pods
- Link and unlink a module for an iOS project with pods
- Link and unlink a module for Android
- Link and unlink assets on iOS and Android

## Related PRs

#17745

## Release Notes

 [INTERNAL] [ENHANCEMENT ] [local-cli/info/link.js] - Make link and unlink more maintainable for all platforms.

### CODE REVIEW

File: local-cli/core/android/index.js [hunk 1]
```diff
@@ -126,3 +126,5 @@ exports.dependencyConfig = function dependencyConfigAndroid(folder, userConfig)
 
   return { sourceDir, folder, manifest, packageImportPath, packageInstance };
 };
+
+exports.linkConfig = require('../../link/android');
```
Comment (rozele) [inline local-cli/core/android/index.js]: Add same for iOS

File: local-cli/link/unlink.js [hunk 2]
```diff
@@ -23,66 +11,36 @@ const promisify = require('./promisify');
...
+    .forEach(platform => {
+      if (!project[platform] || !dependency[platform]) {
+        return;
+      }
 
-  log.info(`Windows module ${packageName} has been successfully unlinked`);
-};
+      const linkConfig = platforms[platform] && platforms[platform].linkConfig && platforms[platform].linkConfig();
+      if (!linkConfig || !linkConfig.isInstalled || !linkConfig.unregister) {
+        return null;
```
Comment (rozele) [inline local-cli/link/unlink.js]: >null [](start = 15, length = 4)

return;

File: local-cli/link/ios/common/unregisterNativeModule.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+module.exports = function registerNativeModule(
+  name,
+  dependencyConfig,
+  params,
+  projectConfig
+) {
+  const isIosInstalled = isInstalledIOS(projectConfig, dependencyConfig);
+  const isPodInstalled = isInstalledPods(projectConfig, dependencyConfig);  
+  if (isIosInstalled) {
+    const allDependencies = getDependencyConfig(config, getProjectDependencies());
```
Comment (rozele) [inline local-cli/link/ios/common/unregisterNativeModule.js]: >getProjectDependencies [](start = 56, length = 22)

need to import

File: local-cli/link/ios/common/unregisterNativeModule.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    const otherDependencies = filter(allDependencies, d => d.name !== packageName);
```
Comment (rozele) [inline local-cli/link/ios/common/unregisterNativeModule.js]: >packageName [](start = 70, length = 11)

name?

File: local-cli/link/android/index.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+exports.linkConfig = function() {
```
Comment (rozele) [inline local-cli/link/android/index.js]: copyrights on new files.

**Comment by grabbou** [comment]
@rozele do you think we should cherry-pick this back to 0.53/0.52/0.51?

**Comment by rozele** [comment]
@grabbou - yes please do cherry-pick back to these versions (and 0.50 if possible), but this PR was closed in favor of #17745.

**Comment by grabbou** [comment]
Yup, got it.

**Pull request closed without merging**